reconstructSPI/iterative_refinement/expectation_maximization.py

```diff
@@ -206,25 +222,29 @@ def iterative_refinement(self, wiener_small_number=0.01, count_norm_const=1):
 
                 for one_slice_idx in range(len(bayes_factors_1)):
                     xyz = xyz_rotated[one_slice_idx]
-                    inserted_slice_3d_r, count_3d_r = IterativeRefinement.insert_slice(
-                        particle_f_deconv_1.real, xyz, n_pix
+                    inserted_slice_3d_r, count_3d_r = insert_slice_v(
+                        particle_f_deconv_1.real, xyz, xyz_voxels
+                    )
+                    inserted_slice_3d_i, count_3d_i = insert_slice_v(
+                        particle_f_deconv_1.imag, xyz, xyz_voxels
                     )
-                    inserted_slice_3d_i, count_3d_i = IterativeRefinement.insert_slice(
-                        particle_f_deconv_1.imag, xyz, n_pix
+                    map_3d_f_updated_1 += np.sum(
```

Comment:
what is the logic of this np.sum ?

Reply:
It gives us the freedom to insert a number of slices in parallel with insert_slice_v (the vectorized insert_slice operation) and then have them summed to form the (non-normalized) map_3d_f_updated.